src/v/cloud_storage/tx_range_manifest.cc

```diff
@@ -12,20 +12,239 @@
 #include <rapidjson/rapidjson.h>
 #include <rapidjson/writer.h>
 
+#include <stdexcept>
+
 namespace cloud_storage {
 
+struct tx_range_manifest_json_handler {
+    using Ch = rapidjson::UTF8<>::Ch;
+
+    // rapidjson's Handler concept.
+    bool Null() { return false; }
+    bool Bool(bool) { return false; }
+    bool Int(int i) { return Int64(i); }
+    bool Uint(unsigned u) { return Int64(u); }
+    bool Int64(int64_t i64) {
+        switch (_state) {
+        case state::expect_version_value:
+            version = static_cast<int>(i64);
+            _manifest_keys |= static_cast<uint8_t>(manifest_keys::version);
+            _state = state::expect_manifest_key;
+            return true;
+        case state::expect_compat_version_value:
+            compat_version = static_cast<int>(i64);
+            if (
+              compat_version
+              > static_cast<int>(tx_range_manifest_version::current_version)) {
+                throw std::runtime_error(fmt::sprintf(
+                  "Can't deserialize tx manifest, supported version {}, "
+                  "manifest "
+                  "version {}, compatible version {}",
+                  static_cast<int32_t>(
+                    tx_range_manifest_version::current_version),
+                  version,
+                  compat_version));
+            }
+            _manifest_keys |= static_cast<uint8_t>(
+              manifest_keys::compat_version);
+            _state = state::expect_manifest_key;
+            return true;
+        case state::expect_range_pid_id_value:
+            ranges.back().pid.id = model::producer_id(i64);
+            _range_keys |= static_cast<uint8_t>(range_keys::pid_id);
+            _state = state::expect_range_key;
+            return true;
+        case state::expect_range_pid_epoch_value:
+            if (
+              i64 < std::numeric_limits<int16_t>::min()
+              || i64 > std::numeric_limits<int16_t>::max()) {
+                return false;
+            }
+            ranges.back().pid.epoch = model::producer_epoch(
+              static_cast<int16_t>(i64));
+            _range_keys |= static_cast<uint8_t>(range_keys::pid_epoch);
+            _state = state::expect_range_key;
+            return true;
+        case state::expect_range_first_value:
+            ranges.back().first = model::offset(i64);
+            _range_keys |= static_cast<uint8_t>(range_keys::first);
+            _state = state::expect_range_key;
+            return true;
+        case state::expect_range_last_value:
+            ranges.back().last = model::offset(i64);
+            _range_keys |= static_cast<uint8_t>(range_keys::last);
+            _state = state::expect_range_end_object;
+            return true;
+        default:
+            return false;
+        }
+    }
+    bool Uint64(uint64_t u64) {
+        if (u64 > std::numeric_limits<int64_t>::max()) {
+            return false;
+        } else {
+            return Int64(static_cast<int64_t>(u64));
```

Comment:
what doe sit mean that `Int64()` type is returned as `bool`?

Reply:
This calls Int64 method which returns a bool.